controllers/rabbitmq/transporturl_controller.go

```diff
@@ -0,0 +1,193 @@
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=transporturls/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=rabbitmq.openstack.org,resources=transporturls/finalizers,verbs=update
+//+kubebuilder:rbac:groups=rabbitmq.com,resources=rabbitmqclusters,verbs=get;list;watch
+//+kubebuilder:rbac:groups=core,resources=secrets,verbs=get;list;watch;create;update;patch;delete;
+
+// Reconcile - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.2/pkg/reconcile
+func (r *TransportURLReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// Fetch the TransportURL instance
+	instance := &rabbitmqv1beta1.TransportURL{}
+	err := r.Client.Get(ctx, req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	helper, err := helper.NewHelper(
+		instance,
+		r.Client,
+		r.Kclient,
+		r.Scheme,
+		r.Log,
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Lookup RabbitmqCluster instance by name
+	labelSelector := map[string]string{
+		"namespace": instance.Namespace,
+		"name":      instance.Spec.RabbitmqClusterName,
+	}
+	rabbit, err := rabbitmq.GetRabbitmqCluster(ctx, helper, instance.Namespace, labelSelector)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// Wait on RabbitmqCluster to be ready
+	rabbitReady := false
+	for _, condition := range rabbit.Status.Conditions {
+		if condition.Reason == "AllPodsAreReady" && condition.Status == "True" {
+			rabbitReady = true
+		}
+	}
+	if !rabbitReady {
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Obtain the default user password from the secret for the above RabbitmqCluster instance
+	// TODO(dprince): Future we may want to use vhosts for each OpenStackService instead.
+	// vhosts would likely require use of https://github.com/rabbitmq/messaging-topology-operator/ which we do not yet include
+	secretName := rabbit.Status.DefaultUser.SecretReference.Name
+	secretNamespace := rabbit.Status.DefaultUser.SecretReference.Namespace
+	defaultUserSecret, _, err := oko_secret.GetSecret(ctx, helper, secretName, secretNamespace)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Extract the password from the secret
+	username, err := base64.StdEncoding.DecodeString(string(defaultUserSecret.Data["username"]))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	password, err := base64.StdEncoding.DecodeString(string(defaultUserSecret.Data["password"]))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	host, err := base64.StdEncoding.DecodeString(string(defaultUserSecret.Data["host"]))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
```

Comment:
I wonder if the secret acquisition here should use the `lib-common` library [1] we have for such things?

[1] https://github.com/openstack-k8s-operators/lib-common/blob/master/modules/common/secret/secret.go

Reply:
thanks @abays